next/components/Molecules/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,111 @@
+import ChevronLeft from '@assets/images/chevron-left.svg'
+import ChevronRight from '@assets/images/chevron-right.svg'
+import cx from 'classnames'
+import React, { useRef } from 'react'
+
+import { VerticalCardButton } from '../../HomePage/VerticalCardButton'
+
+interface SectionPromosProps {
+  listClassName?: string
+  itemClassName?: string
+  items: { element: JSX.Element | null; key: string | undefined }[]
+  shiftIndex?: number
+  visibleItems?: number
+}
+
+const Carousel = ({
+  listClassName,
+  itemClassName,
+  items,
+  shiftIndex = 1,
+  visibleItems = 3,
+}: SectionPromosProps) => {
+  const [currentItem, setCurrentItem] = React.useState(0)
+
+  const scrollerRef = useRef<HTMLUListElement>(null)
+
+  const scrollToImage = (i: number, instant = false) => {
+    setCurrentItem(i)
+    if (!scrollerRef.current) return
+    const offset = (scrollerRef.current.scrollWidth / items.length) * i
+
+    scrollerRef.current?.scroll({
+      left: offset,
+      behavior: instant ? 'auto' : 'smooth',
+    })
+  }
+
+  const nextImage = () => {
+    const nextIdx = currentItem + shiftIndex
+    const maxIdx = items.length - visibleItems
+    if (nextIdx >= maxIdx + shiftIndex) {
+      // If on last page, go to 0
+      // scrollToImage(0, true)
+      return
+    }
+
+    if (nextIdx > Math.min(maxIdx - shiftIndex, maxIdx - visibleItems)) {
+      scrollToImage(nextIdx)
+      return
+    }
+
+    scrollToImage(nextIdx)
+  }
+
+  const previousImage = () => {
+    scrollToImage(currentItem - shiftIndex)
+  }
+
+  const sliderControl = (isLeft: boolean) =>
+    items.length >= visibleItems ? (
+      <VerticalCardButton
+        onPress={isLeft ? previousImage : nextImage}
+        size="large"
+        className={cx('absolute top-0 bottom-0 z-10 my-auto', {
+          'left-0 -translate-x-1/2 transform': isLeft,
+          'right-0 translate-x-1/2 transform': !isLeft,
+          hidden:
+            (isLeft && currentItem === 0) ||
+            (!isLeft && currentItem + shiftIndex >= items.length - visibleItems + shiftIndex),
```

Comment:
This is hardly readable, please add a comment explaining the maths here :D

Reply:
For me it is questionable if this should be the same component for left and right arrow  if it requires "isLeft" condition everywhere.

Reply:
IMHO it's better to keep logic inside one component, close to each other, then spread it across code. What this provide is when code needs to be edited there is only one single source of truth and you will change both behaviour in place. In case there is separate behaviour in more places, often happens that code is changed in one part and forgotten in second, third... Comment with math explanation will be provided.